### PR TITLE
Add nullability, generic annotations for better swift compatibility

### PR DESCRIPTION
**Pull request by ytok**
Updated the podspec too.

### CODE REVIEW

File: SimpleAL.podspec [hunk 1]
```diff
@@ -1,10 +1,10 @@
 Pod::Spec.new do |s|
   s.name         = 'SimpleAL'
-  s.version      = '1.0.1'
+  s.version      = '2.0.0'
```
Comment (SSheldon) [inline SimpleAL.podspec]: Oh didn't expect this to require a major version bump, how does adding these annotations break things?

File: Classes/NSLayoutConstraint+SimpleAL.h [hunk 2]
```diff
@@ -21,3 +23,5 @@
 + (NSLayoutConstraint *)al_constrainViewProperty:(SimpleALViewProperty *)viewProperty1 greaterThanOrEqualToValue:(CGFloat)value;
```
Comment (SSheldon) [inline Classes/NSLayoutConstraint+SimpleAL.h]: While we're at it, should these return `instancetype`?

File: SimpleAL.podspec (hunk 1 above)
Comment (ytok) [inline SimpleAL.podspec]: Err I meant to change this, thanks.

**Comment by SSheldon** [comment]
Ship it! :shipit:

**Pull request merged**